spinalcordtoolbox/moco.py

```diff
@@ -853,3 +856,128 @@ def spline(folder_mat, nt, nz, verbose, index_b0 = [], graph=0):
 
     sct.printv('\n...Done. Patient motion has been smoothed', verbose)
     sct.printv('------------------------------------------------------------------------------\n', verbose)
+
+def split_to_odd_and_even(param):
+    """
+    Split data into two datasets (even and odd slices) and run moco separately, then merge data back together
+    :param param: ParamMoco class
+    :return: None
+    """
+
+    orig_name = param.fname_data
+    file_data = 'data.nii'  # corresponds to the full input data (e.g. dmri or fmri)
+    file_mask = 'mask.nii'
+    file_data_dirname, file_data_basename, file_data_ext = sct.extract_fname(file_data)
+    file_data_even = 'data_even.nii'
+    file_data_odd = 'data_odd.nii'
+    file_mask_even = 'mask_even.nii'
+    file_mask_odd = 'mask_odd.nii'
+    file_data_merged = 'data_merged.nii'
+
+    # Start timer
+    start_time = time.time()
+
+    path_tmp = sct.tmp_create(basename="moco", verbose=param.verbose)
+
+    # Copying input data to tmp folder
+    sct.printv('\nCopying input data to tmp folder and convert to nii...', param.verbose)
+    convert(param.fname_data, os.path.join(path_tmp, file_data))
+    if param.fname_mask != '':
+        convert(param.fname_mask, os.path.join(path_tmp, file_mask), verbose=param.verbose)
+        # Update field in param (because used later in another function, and param class will be passed)
+        param.fname_mask = file_mask
+
+    # Build absolute output path and go to tmp folder
+    curdir = os.getcwd()
+    path_out_abs = os.path.abspath(param.path_out)
+    os.chdir(path_tmp)
+
+    # Load data and get its dimensions
+    sct.printv('\nGet dimensions of data...', param.verbose)
+    im_data = Image(file_data)
+    nx, ny, nz, nt, px, py, pz, pt = im_data.dim
+    sct.printv('  ' + str(nx) + ' x ' + str(ny) + ' x ' + str(nz) + ' x ' + str(nt), param.verbose)
+
+    # Split 4D data along z dimension
+    #sct.printv('\nSplit along z dimension...', param.verbose)
+    im_data_split_list = split_data(im_data, 2)  # 0 - x, 1 - y, 2 - z, 3 - t
```

Comment:
you want to split along the SI direction. Dimension 2 does not always correspond to SI. E.g., if your orientation is AIL, this code will split along the right-left direction.